metadata_backend/helpers/validator.py

```diff
@@ -52,33 +56,40 @@ def resp_body(self) -> str:
             LOG.exception("Submitted file does not not contain valid XML syntax.")
             return ujson.dumps({"isValid": False, "detail": {"reason": reason, "instance": instance}})
 
-        except XMLSchemaValidationError as error:
-            # Parse reason and instance from the validation error message
-            reason = str(error.reason)
-            response: Dict = {"isValid": False, "detail": {"reason": reason}}
-            if error.elem:
-                instance = etree_tostring(error.elem, encoding="unicode")
-                # Replace element address in reason with instance element
-                if "<" in reason and ">" in reason:
-                    instance_parent = "".join((instance.split(">")[0], ">"))
-                    reason = re.sub("<[^>]*>", instance_parent + " ", reason)
-                    response["detail"]["reason"] = reason
-                response["detail"]["instance"] = instance
-
-            LOG.exception("Submitted file is not valid against schema.")
-            return ujson.dumps(response)
-
-        except URLError as error:
-            reason = f"Faulty file was provided. {error.reason}."
-            LOG.error(reason)
-            raise web.HTTPBadRequest(reason=reason)
-
     def _parse_error_reason(self, error: ParseError) -> str:
-        """Generate better error reason."""
+        """Generate better error reason for ParseError."""
         reason = str(error).split(":", maxsplit=1)[0]
         position = (str(error).split(":")[1])[1:]
         return f"Faulty XML file was given, {reason} at {position}"
 
+    def _format_xml_validation_error_reason(self, errors: List) -> Dict:
+        """Generate the response json object for validation error(s)."""
+        response: Dict = {"isValid": False, "detail": []}
+        found_lines = []
+        for error in errors:
+            reason = str(error.reason)
+            instance = str(error.path)
+
+            # Add line number to error reason
+            lines = self.xml_content.split("\n")
+            elem_name = error.obj[error.index].tag if isinstance(error, XMLSchemaChildrenValidationError) else error.obj
+            for (i, line) in enumerate(lines, 1):
+                if elem_name in line and i not in found_lines:
+                    line_num = i
+                    found_lines.append(i)
+                    break
+            if re.match(r"^.*position [0-9]+", reason):
+                # line number replaces element position which is more valuable information
+                reason = re.sub(r"position [0-9]+", f"line {line_num}", reason)
+            else:
+                # line number still added as extra info to error reason
+                reason = reason + f" (line {line_num})"
+            detail = {"reason": reason, "instance": instance}
+            response["detail"].append(detail)
+
+        response["detail"] = response["detail"][0] if len(response["detail"]) == 1 else response["detail"]
+        return response
```

Comment:
The idea here is to include all the failing instances within the file in the same response as a list since it's possible to collect them all. Not sure if that's needed tho, but in case this is a good idea the openapi docs might need to be updated since currently the `response["detail"]` type is set as object. As I was going through the `specification.yml` I noticed there's existing unrelated error there on line 55. So I'll leave the doc update out of this for now.

Reply:
Given the other comment about the `detail` field, what do you think, could `detail` be a string? Likely those massages will not be parsed by a script, and even if they are, a regex could do the trick in extracting information.

Reply:
Now that I realize I might be asking pretty much for a new feature, that error messages stick to the JSON Problem Details format. But what do you think, does it make sense in this case?